bank/api/api.go

```diff
@@ -0,0 +1,246 @@
+		account.UserID = session.UserID
+
+		_, err := accountStore.InsertAccount(account)
+		if err != nil {
+			errorResponse(w, http.StatusInternalServerError, "Create User Account Failed", "error", "unexpected_error", err.Error())
+			return
+		}
+
+		w.WriteHeader(http.StatusCreated)
+	})
+}
+
+func DeleteUserAccount(accountStore domain.AccountStore) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		session := context.Get(r, "session").(*domain.Session)
+
+		var account domain.Account
+
+		value := r.URL.Query()
+		currency := value["currency"]
+
+		if len(currency) == 0 {
+			errorResponse(w, http.StatusBadRequest, "Delete User Account Failed", "request", "URL_parameters", "need_currency_to_be_specified_in_URL")
+			return
+		}
+		account.UserID = session.UserID
+		account.Currency = currency[0]
+
+		result, err := accountStore.GetAccount(account.UserID, account.Currency)
+		if err != nil {
+			errorResponse(w, http.StatusInternalServerError, "Delete User Accounts Failed", "error", "unexpected_error", err.Error())
+			return
+		}
+
+		err = accountStore.DeleteAccount(account.UserID, result.AccountID, account.Currency)
+		if err != nil {
+			errorResponse(w, http.StatusInternalServerError, "Delete User Accounts Failed", "error", "unexpected_error", err.Error())
+			return
+		}
+	})
+}
+
+func UserAccountDeposit(accountStore domain.AccountStore) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		session := context.Get(r, "session").(*domain.Session)
+
+		var account domain.Account
+		json.NewDecoder(r.Body).Decode(&account)
+		account.UserID = session.UserID
+
+		err := accountStore.Deposit(account)
+		if err != nil {
+			errorResponse(w, http.StatusInternalServerError, "User Account Deposit Failed", "error", "unexpected_error", err.Error())
+			return
+		}
+	})
+}
+
+func UserAccountWithdraw(accountStore domain.AccountStore) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		session := context.Get(r, "session").(*domain.Session)
+
+		var account domain.Account
+		json.NewDecoder(r.Body).Decode(&account)
+		account.UserID = session.UserID
+
+		err := accountStore.Withdraw(account)
+		if err != nil {
+			switch err {
+			case domain.ErrWithdrawMoreThanHave:
+				errorResponse(w, http.StatusBadRequest, "User Account Withdraw Failed", "account", "money_availability", "cannot_withdraw_more_than_have_available")
+				return
+			default:
+				errorResponse(w, http.StatusInternalServerError, "User Account Withdraw Failed", "error", "unexpected_error", err.Error())
+				return
+			}
+		}
+	})
+}
+
+func UserTransactionHistory(accountStore domain.AccountStore) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		session := context.Get(r, "session").(*domain.Session)
+
+		var historyRequest domain.History
+		json.NewDecoder(r.Body).Decode(&historyRequest)
+
+		historyRequest.UserID = session.UserID
```

Comment:
GetHistory(historyRequest) 
or
GetHistory(session.UserID, accountID) // accountID could be passed as ?accountID=1234